### ISSUE TITLE
[Bug]: Accessibility - In multi-select drop down, NVDA does not announce placeholder text when items selected

### ISSUE THREAD

**Issue by binhphan009** [microsoft/fluentui#29095]
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
    Memory: 2.77 GB / 15.73 GB
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (116.0.1938.69)
    Internet Explorer: 11.0.19041.1566
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codesandbox.io/s/dropdown-multi-select-ljzjlz?file=/index.js

## Actual Behavior
Repro Steps:

1. Open NVDA
2. Focus on to Multi select
3. Select 1 item (Robin Counts) in the DropDown
4. Screen reader announce "Robin Counts selected,... **menu button subMenu**". But the Placeholder or the Dropdown name/label was not announced.
<img width="458" alt="image" src="https://github.com/microsoft/fluentui/assets/105278576/9201a642-4b5a-45a1-9a20-94e3e20b8d6e">

5. Tab outside to another element. Then, go backward using "Shift + Tab". Screen reader announce "clickable **menu button subMenu**"

<img width="482" alt="image" src="https://github.com/microsoft/fluentui/assets/105278576/f6f33ede-a83e-4c17-9b8e-c73a511e97d0">

## Expected Behavior
When the multi-select dropdown becomes focused, the placeholder string should be announced.
E.g:  **Select your heroes, collapsed, menu button subMenu**

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
